packages/@aws-cdk/aws-ec2/lib/vpc-endpoint-service.ts

```diff
@@ -0,0 +1,87 @@
+import { ArnPrincipal } from '@aws-cdk/aws-iam';
+import { Construct, IResource, Resource } from '@aws-cdk/core';
+import { CfnVPCEndpointService, CfnVPCEndpointServicePermissions } from './ec2.generated';
+
+/**
+ * A VPC endpoint service.
+ */
+export interface IVpcEndpointService extends IResource {
+}
+
+/**
+ * A VPC endpoint service
+ * @resource AWS::EC2::VPCEndpointService
+ */
+export class VpcEndpointService extends Resource implements IVpcEndpointService {
+
+  /**
+   * One or more network load balancer ARNs to host the service.
+   * @attribute
+   */
+  public readonly networkLoadBalancerArns: string[];
+
+  /**
+   * Whether to require manual acceptance of new connections to the service.
+   * @attribute
+   */
+  public readonly acceptanceRequired: boolean;
+
+  /**
+   * One or more Principal ARNs to allow inbound connections to.
+   * @attribute
+   */
+  public readonly whitelistedPrincipals: ArnPrincipal[];
+
+  private readonly endpointService: CfnVPCEndpointService;
+
+  constructor(scope: Construct, id: string, props: VpcEndpointServiceProps) {
+    super(scope, id);
+
+    if (props.networkLoadBalancerArns === undefined || props.networkLoadBalancerArns.length === 0) {
```

Comment:
I *think* the Typescript way to do this is `if !props.networkLoadBalancerArns`.

Reply:
@rhboyd `!props.networkLoadBalancerArns` and `props.networkLoadBalancerArns === undefined` do the same thing (given the current types! this is not always true!) but the second check, to make sure that `props.networkLoadBalancerArns.length === 0` is also important.

An empty array would check as *truthy*, not *falsy*.